incipyt/project/structure.py

```diff
@@ -1,147 +1,13 @@
-
-    All iterative methods and in operator consider only `confirmed` variables.
-    """
-
-    def clear(self):
-        self._confirmed.clear()
-        self.data.clear()
-        self.data.update(os.environ.copy())
-
-        self.data.setdefault("PYTHON_CMD", sys.executable)
-        self._confirmed.add("PYTHON_CMD")
-        self.data.setdefault("YEAR", date.today().year)
-        self._confirmed.add("YEAR")
-
-    def __init__(self):
-        self._confirmed = set()
-        self.data = {}
-        self.clear()
-
-    def __getitem__(self, key):
-        if key not in self._confirmed:
-            logger.debug("Missing environ variable %s, request it.", key)
-            self.data[key] = self._prompt(key)
-            self._confirmed.add(key)
-
-        return self.data[key]
-
-    def __setitem__(self, key, env_value):
-        if not isinstance(env_value, EnvValue):
-            env_value = EnvValue(env_value)
-
-        if key in self.data and not env_value.update:
-            raise ValueError(f"Environ variable {key} already exists, use update.")
-
-        logger.debug("Set environ variable %s=%s.", key, env_value.value)
-        self.data[key] = env_value.value
-        if env_value.confirmed and key not in self._confirmed:
-            self._confirmed.add(key)
-
-    def setdefault(self, key, value):
-        if key in self:
-            return self.data[key]
-        self[key] = value
-        return self.data[key]
-
-    def __iter__(self):
-        return iter(self._confirmed)
-
-    def __contains__(self, key):
-        return key in self._confirmed
-
-    def keys(self):
-        return self._confirmed
-
-    def values(self):
-        return [self.data[key] for key in self._confirmed]
-
-    def items(self):
-        return [(key, self.data[key]) for key in self._confirmed]
-
-    def getitems_sanitized(self, keys, sanitizer=None):
-        """Get multiple items at once and sanitize them.
-
-        See also :func:`incipyt.system.Environment.__getitem__`, which will be
-        used to pull each key from the environment.
-
-        :param keys: Required environment keys. If a key is `None`, it will be ignored.
-        :type keys: :class:`collections.abc.Sequence`
-        :param sanitizer: Will be called on key-value pairs to sanitize values.
-        :type sanitizer: :class:`function`
-        :return: Sanitized environment key-value pairs.
-        :rtype: :class:`dict`
-        """
-        return {
-            key: sanitizer(key, self[key]) if sanitizer else self[key]
-            for key in keys
-            if key is not None
-        }
-
-    def _prompt(self, key):
-        return click.prompt(
-            key.replace("_", " ").lower().capitalize(),
-            default=self.data[key] if key in self.data else "",
-            type=str,
-        )
-
-    def __ior__(self, other):
-        self.update(other)
-        return self
-
-
-environ = _Environ()
-
-
-class _Structure:
+class Structure:
```

Comment:
Why remove leading _ while you import it later with `as _` ?

Reply:
I don't like private classes (hence no `_`) but I didn't want to break compatibility (hence the import as)

Reply:
oh but that's singleton actually :facepalm: I should had kept the private class